app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedConstructionForm.kt

```diff
@@ -0,0 +1,36 @@
+package de.westnordost.streetcomplete.quests.construction
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AYesNoQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.OtherAnswer
+import kotlinx.android.synthetic.main.quest_buttonpanel_yes_no.*
+import java.util.*
+
+class MarkCompletedConstructionForm : AYesNoQuestAnswerFragment<CompletedConstructionAnswer>() {
+    override val otherAnswers = listOf(
+        OtherAnswer(R.string.quest_construction_completes_at_known_date) { setFinishDate() }
+    )
+
+    private fun setFinishDate() {
+        val tomorrow = Calendar.getInstance()
+        tomorrow.add(Calendar.DAY_OF_MONTH, 1)
+        val year = tomorrow.get(Calendar.YEAR)
+        val month = tomorrow.get(Calendar.MONTH)
+        val day = tomorrow.get(Calendar.DAY_OF_MONTH)
+        val dpd = DatePickerDialog(requireContext(), { _, yearSelected, monthSelected, dayOfMonthSelected ->
+            applyAnswer(OpeningDateAnswer(Date(yearSelected-1900, monthSelected, dayOfMonthSelected)))
```

Comment:
this Date-constructor is deprecated, see https://docs.oracle.com/javase/7/docs/api/java/util/Date.html#Date(int,%20int,%20int)